python/pdstools/reports/ModelReport.qmd

```diff
@@ -197,65 +328,135 @@ predictors. This can be configured via a parameter to this report.
+    )
 
-    display(Markdown(": A Great Predictor {.striped .hover}"))
-    
-    # In the R version I managed to use a table to format it so
-    # the description is on the left, the plots on the right. With
-    # quarto I couldnt figure out, so the plots are seperate. Also
-    # the width/height of the figures doesnt seem changeable,
-    # show(width=20, height=10) has no effect
+    show(predictor_properties.collect().to_pandas())
 
     fig = datamart.plotPredictorBinning(modelids=[model_id], predictors=[pred])
 
     # Customize some of the default styling of the plot
-    # TODO except for title consider moving into library
-    # TODO make the fly-over better
-    fig.update_layout(title=pred, xaxis_title="")
+
+    fig.update_layout(title="Predictor Binning", xaxis_title="")
+    # fig.layout.yaxis2.tickformat = ",.3%"
     fig.data[0].opacity=0.5
     fig.data[1].line.color="#EF8B08"
     fig.data[1].line.width=3
     fig.data[1].marker.color="black"
+
+    # Add line for base propensity, TODO consider putting back in the library
+    fig.add_hline(y=base_propensity, line_dash="dash", line_color="grey", yref='y2')
+    # fig.update_xaxes(type='category') # prevent plotly from trying to guess
+
+    fig.update_layout(width=700, height=300)
+    fig.show()
+
+    # "Philip Mann" plot with simple red/green lift bars relative to base propensity
+    # TODO move below "Philip Mann" plot into a plot function (plotBinningLift)
+
+    pm_plot_binning_table = (
+        predictor_data
+        .select(
+            pl.col(["BinSymbol", "BinPositives", "BinIndex"]), 
+            (lift(pl.col("BinPositives"), pl.col("BinNegatives")) - 1.0)
+        )
+        .with_columns(
+            pl.when((pl.col("Lift") >= 0.0) & (pl.col("BinPositives") > 5)).then(pl.lit('pos'))
+            .when((pl.col("Lift") >= 0.0) & (pl.col("BinPositives") <= 5)).then(pl.lit('pos_shaded'))
+            .when((pl.col("Lift") < 0.0) & (pl.col("BinPositives") > 5)).then(pl.lit('neg'))
+            .otherwise(pl.lit('neg_shaded'))
+            .alias("Direction"))
+        .sort("BinIndex")
+        .collect()
+    )
+
+    fig = px.bar(
+        x=pm_plot_binning_table['Lift'],
+        y=pm_plot_binning_table['BinSymbol'], 
+        color=pm_plot_binning_table['Direction'],
+        color_discrete_map = {
+            'neg': '#A01503', 
+            'pos': '#5F9F37',
+            'neg_shaded' : '#DAA9AB',
+            'pos_shaded' : '#C5D9B7'},
+        orientation='h',
+        template='pega'
+    )
+    fig.update_yaxes(type='category')
+    fig.layout.xaxis.tickformat = ",.2%"
+    fig.add_vline(x=0, line_color="black")
+    fig.update_layout(
+        yaxis = dict(autorange="reversed"),
+        title="Propensity Lift", xaxis_title="", yaxis_title="")
+    # Not convinced why we need below. It should respect the binindex order but w/o this it doesnt always.
+    fig.update_yaxes(categoryorder='array', categoryarray=pm_plot_binning_table['BinSymbol'])
```

Comment:
I _think_, if there's no missing bin, it tries to cast to numeric and tries to do magic with sorting. If there's a missing bin, then casting to numeric will fail and it will just keep the order